CONTRIBUTING.md

```diff
@@ -3,16 +3,57 @@
 We're always looking for your help to fix bugs and improve the product. Create a pull request and we'll be happy to take a look.
 Start by reading the [Engineering Design](docs/HighLevelDesign.md). You can find the doxygen generated documentation [here](https://microsoft.github.io/onnxruntime/).
 
-# Checkin procedure
-1. Fork the repo
-2. git clone your fork
-3. Create feature branch
-4. Make and checkin your changes along with unit tests
-5. git commit your changes
-6. git push origin HEAD
-7. To request merge into master send a pull request from the web ui
-https://github.com/Microsoft/onnxruntime.
-8. Add 'Microsoft/onnxruntime' as a reviewer.
+# Proposing new public APIs
+The ONNX runtime has a collection of [public API's](docs/HighLevelDesign.md).  Some of these API's make their way back into the Windows OS.  We make compatability committments for these API's and follow a structured process when adding to them.  Please use the [Feature Request issue template](issues/new?template=feature_request.md) before starting any PR's that affect any of the public API's.
+
+# Process details
+
+Please search the [issue tracker](https://github.com/microsoft/onnxruntime/issues) for a similar idea first: there may already be an issue you can contribute to.
+
+1. **Create Issue**  
+All code changes must be tied to an Issue. 
+To propose a new feature or API please start by filing a new issue in the [issue tracker](https://github.com/microsoft/onnxruntime/issues).  
+Include as much detail as you have. It's fine if it's not a complete design: just a summary and rationale is a good starting point.
+
+2. **Wait for Team Owner**  
+We will assign an ONNX Runtime team owner to your issue. The ONNX Runtime team regularly triages all incoming issues.  
+
+3. **Discussion**  
+We'll keep the issue open for community discussion until the team owner decides it's ready or should be closed.  
+Note that if an issue isn't a high priority or has many open questions then it might stay open for a long time.
+
+4. **Owner Review**  
+The ONNX Runtime team will review the proposal and either approve or close the issue based on whether it broadly aligns with the [Onnx Runtime Roadmap](../docs/Roadmap.md) and contribution guidelines.
```

Comment:
specifically, the "High Level Goals" section of the roadmap. the investments mentioned in the rest of the doc refer to what the core team is working on, but we welcome contributions to other areas that may not be the core team's focus. this is a main value of having a community! :)

Reply:
Added the specific section to it.